Universal FE Randomizer/src/fedata/gba/fe6/FE6Data.java

```diff
@@ -2897,4 +2906,101 @@ public int bytesPerAnimation() {
 	public GBAFESpellAnimationCollection spellAnimationCollectionAtAddress(byte[] data, long offset) {
 		return new FE6SpellAnimationCollection(data, offset);
 	}
+
+	@Override
+	public long portraitDataTableAddress() {
+		return 0x66074C;
+	}
+
+	@Override
+	public int numberOfPortraits() {
+		return 231 - 2;
+	}
+
+	@Override
+	public int bytesPerPortraitEntry() {
+		return 16;
+	}
+
+	@Override
+	public GBAFEPortraitData portraitDataWithData(byte[] data, long offset, int faceId) {
+		return new FE6PortraitData(data, offset, faceId, false);
+	}
+	
+	
+	public static Map<Integer, List<Integer>> faceIdRelationMap = createFaceIdRelationMap();
+	private static Map<Integer, List<Integer>> createFaceIdRelationMap(){
+		Map<Integer, List<Integer>> relationMap = new HashMap<>();
+		// FE6 doesn't have any
+		return relationMap;
+	}
+
+	@Override
+	public List<Integer> getRelatedPortraits(Integer faceId) {
+		List<Integer> result = faceIdRelationMap.get(faceId);
+		return result == null ? new ArrayList<>() : new ArrayList<>(result);
+	}
+	
+	public static Map<Integer, List<Integer>> nameIndexRelationMap = createNameIndexRelationMap();
+	private static Map<Integer, List<Integer>> createNameIndexRelationMap(){
+		Map<Integer, List<Integer>> relationMap = new HashMap<>();
+		relationMap.put(0x7F1, Arrays.asList(0x7F2)); // Dayan
+		relationMap.put(0x7F6, Arrays.asList(0x7F7)); // Douglas
+		relationMap.put(0x7FD, Arrays.asList(0x7FE)); // Yoder
+		relationMap.put(0x813, Arrays.asList(0x814)); // Niime
+		relationMap.put(0x817, Arrays.asList(0x818)); // Juno
+		relationMap.put(0x819, Arrays.asList(0x81A, 0x81B)); // Thea
+		relationMap.put(0x823, Arrays.asList(0x824)); // Gonzalez
+		relationMap.put(0x828, Arrays.asList(0x829)); // Echidna
+		relationMap.put(0x82B, Arrays.asList(0x82C)); // Geese
+		
+		return relationMap;
+	}
+	
+	@Override
+	public List<Integer> getRelatedNames(Integer nameIndex) {
+		List<Integer> result = nameIndexRelationMap.get(nameIndex);
+		return result == null ? new ArrayList<>(): new ArrayList<>(result);
+	}
+	
+	@Override
+	public int getHuffmanTreeStart() {
+		return 0x6E0;// Resolved once
+	}
+
+	@Override
+	public int getHuffmanTreeEnd() {
+		return  0x6DC;// Resolved twice
+	}
+
+	@Override
+	public int getTextTablePointer() {
+		return  0x13B10;
+	}
+
+	@Override
+	public int getNumberOfTextStrings() {
+		return 0xD0D;
+	}
+	
+	private final Set<Integer> excludedIndicies = generateExcludedIndiciesSet();
+	
+	@Override
+	public Set<Integer> getExcludedIndiciesFromNameUpdate() {
+		return excludedIndicies;
+	}
+
+	private Set<Integer> generateExcludedIndiciesSet() {
+		Set<Integer> indicies = new HashSet<>();
+		indicies.add(0x736); // Lancereaver
```

Comment:
Lancereaver doesn't need it technically, since the text swapping should be looking for word boundaries and not just an occurrence of the same sequence of characters. It should be why "Sain" doesn't change instances of "Saint".

Reply:
You're right, though I'll leave them there just for consistency